components/gfx/paint_context.rs

```diff
@@ -909,12 +909,37 @@ impl<'a> PaintContext<'a> {
         }
 
         // FIXME(pcwalton): This surface might be bigger than necessary and waste memory.
-        let size = self.draw_target.get_size();
-        let size = Size2D(size.width, size.height);
+        let size = self.draw_target.get_size(); //Az size.
+        let mut size = Size2D(size.width, size.height); //Geom::Size.
+
+        // Pre-calculate if there is a blur expansion need.
+        let accum_blur = filters::calculate_accumulated_blur(filters);
+        let mut matrix = Matrix2D::identity();
```

Comment:
Why not just use `let mut matrix = self.draw_target.get_transform();`? Then you wouldn't need the second `if accum_blur > Au(0) { ...` below.